### ISSUE TITLE
https portal

### ISSUE THREAD

**Issue by Nimo11** [tzapu/WiFiManager#1044]
Hello,

I will find it very useful to be able to use the portal in https. I use the non-blocking portal and it would be more secure. Do you have it in your road map.

Thank you for this very useful work

**Comment by tablatronix** [comment]
Yes ideally you would provide your own websever and pas it into wm

**Comment by Nimo11** [comment]
You could maybe change handleRequest to this :

```
void WiFiManager::handleRequest() {
  if (WiFiManager::AuthenticationRequired&&!server->authenticate(WiFiManager::AuthUser,WiFiManager::AuthPassword))
  {
    server->requestAuthentication(HTTPAuthMethod::DIGEST_AUTH);
  }
  _webPortalAccessed = millis();
}
```

**Comment by tablatronix** [comment]
What’s that do?

**Comment by Nimo11** [comment]
It add authentication ability. 

Changing handleRequest with this code and adding in header file this property :

  bool AuthenticationRequired;
  const char* AuthUser;
  const char* AuthPassword;

So developers can decide to ask authentication simply.

Sorry for this too fast reply.

**Comment by tablatronix** [comment]
Does this work with captive portal ?

**Comment by Nimo11** [comment]
Yes, all requests are processed. But the behavior can be modified by changing the value of AuthenticationRequired after the autoconnect

**Comment by tablatronix** [comment]
Ok this definitely does not work for captive portals. I am gonna try to add a landing page and see if that works.

**Comment by Nimo11** [comment]
I try it this week and it work me with dev branch. I restart with acces point and i have login needed, After configuration i have a login page on my portal. What's wrong with your test?

**Comment by tablatronix** [comment]
Https authenticaiton does not work on my devices. It never pops up
